### PR TITLE
fix(switch): visual overflow with longer translation strings (FE-6374)

### LINKED ISSUE #6869: Label and switch vertical alignment do not match for https://carbon.sage.com/?path=/story/switch--with-label-inline
### Description

When using the switch with labelInline component (or labelInline and reversed), the vertical alignment is off between the label text and the switch itself. 

image

### Reproduction

https://carbon.sage.com/?path=/story/switch--with-label-inline

### Steps to reproduce

_No response_

### JIRA ticket numbers (Sage only)

_No response_

### Suggested solution

Adding a height prop resolves this so presumably a height adjustment is all that is needed on the component. 

### Carbon version

v141.2.1

### Design tokens version

_No response_

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Additional context

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by nicktitchmarsh**
Fixes FE-6374
fix #6869

### Proposed behaviour

Switch should adapt to varying string lengths

### Current behaviour

Longer translation strings cause text to overflow due to fixed width of switch component

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Run the storybook instance
Edit the pl-pl.ts file to make the translations longer
Set the locale in the docs page.

### CODE REVIEW

File: src/components/switch/__internal__/switch-slider.style.ts [hunk 1]
```diff
@@ -30,103 +30,108 @@ const StyledSwitchSlider = styled.span`
...
+        `}
       `
-        background-color: var(--colorsActionDisabled500);
-        border-color: var(--colorsActionMinorTransparent);
+    }
+
+    ${
+      size === "large" &&
+      css`
+        ${theme?.roundedCornersOptOut ? "border-radius: 30px;" : ""}
```
Comment (edleeks87) [inline src/components/switch/__internal__/switch-slider.style.ts:107]: question (non-blocking): do we actually need to set this I don't notice any visual changes when I unset this in the dev console?

File: src/components/switch/__internal__/switch-slider.style.ts [hunk 2]
```diff
@@ -30,103 +30,108 @@ const StyledSwitchSlider = styled.span`
...
+      css`
+        background-color: var(--colorsActionMinor500);
+        border-color: var(--colorsActionMinorTransparent);
 
-      &::before {
-        transform: translateX(36px);
-        background-color: var(--colorsActionMinorYang100);
-      }
+        &::before {
+          margin-left: calc(100% - ${size === "large" ? "40px" : "24px"});
```
Comment (edleeks87) [inline src/components/switch/__internal__/switch-slider.style.ts:68]:
```suggestion
          margin-left: calc(100% - ${size === "large" ? "var(--spacing500)" : "var(--spacing300)"});
```

File: src/components/switch/__internal__/switch-slider.style.ts [hunk 3]
```diff
@@ -30,103 +30,108 @@ const StyledSwitchSlider = styled.span`
...
     box-sizing: border-box;
+    margin-top: ${size === "large" ? "-47px" : "-28px"};
+    align-items: center;
 
     &::before {
       background-color: var(--colorsActionMinor400);
       bottom: 4px;
       content: "";
-      height: 16px;
+      height: ${size === "large" ? "32px" : "16px"};
```
Comment (edleeks87) [inline src/components/switch/__internal__/switch-slider.style.ts:52]:
```suggestion
      height: ${size === "large" ? "var(--spacing400)" : "var(--spacing200)"};
```

File: src/components/switch/__internal__/switch-slider.style.ts [hunk 4]
```diff
@@ -30,103 +30,108 @@ const StyledSwitchSlider = styled.span`
...
       position: absolute;
       left: 4px;
-      transition: transform 0.4s;
-      width: 16px;
+      transition: margin-left 0.4s;
+      width: ${size === "large" ? "32px" : "16px"};
```
Comment (edleeks87) [inline src/components/switch/__internal__/switch-slider.style.ts:56]:
```suggestion
     width: ${size === "large" ? "var(--spacing400)" : "var(--spacing200)"};
```

File: src/components/switch/__internal__/switch-slider.style.ts [hunk 5]
```diff
@@ -30,103 +30,108 @@ const StyledSwitchSlider = styled.span`
...
-      ? "90px"
-      : "var(--borderRadius400)"};
+    border-radius: ${
+      theme?.roundedCornersOptOut ? "90px" : "var(--borderRadius400)"
+    };
     border-style: solid;
     border-color: var(--colorsActionMinor400);
     border-width: var(--borderWidth200);
     box-sizing: border-box;
+    margin-top: ${size === "large" ? "-47px" : "-28px"};
```
Comment (edleeks87) [inline src/components/switch/__internal__/switch-slider.style.ts:45]: comment (non-blocking): I think this is needed as we're setting the height on 3 elements. We can probably get rid of this if we were to only apply the height to `StyledSwitchSlider` when size is large and also setting the input to have the following:

```
input {
    position: absolute;
    height: fit-content;
  }

**Review by edleeks87 — COMMENTED** [review]
I think it might be useful to include a test story that demonstrates what these changes were trying to address. Wrt the rest of my comments, I'm happy to have all that addressed in a separate ticket if this is time sensitive

**Review by nineteen88 — APPROVED** [review]

**Review by Parsium — APPROVED** [review]
**comment**: we might consider replacing the "type" attribute on `StyledSwitchSlider` with a CSS class in the future.

Although the "type" attribute is not semantically meaningful on a `<div>` element, it appears macOS VoiceOver (in Safari) uses it to describe the input when focused:

Screenshot 2024-07-29 at 11 39 29

**Review by Parsium — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Pull request merged**